### ISSUE TITLE
dotnet test should forward msbuild errors through standard error stream

### ISSUE THREAD

**Issue by RikkiGibson** [microsoft/vstest#2706]
## Steps to reproduce
```pwsh
# in Powershell

# try to run tests on a nonexistent dll. we get error output, yay!
> dotnet test 'foo.dll' | out-null    
The test source file "C:\Users\rikki\src\roslyn\foo.dll" provided was not found.

# now introduce additional quotes in the filename, perhaps due to some mishap involving quoting and shell escaping.
# there is no output at all from this command :(
> dotnet test '''foo.dll''' | out-null

# the reason is that the msbuild errors are given through standard output
> dotnet test '''foo.dll'''
MSBUILD : error MSB1009: Project file does not exist.
Switch: 'foo.dll'
```

## Expected behavior
When the dotnet test command has errors, relevant error messages should be given through the standard error stream, not standard output

## Actual behavior
Sometimes dotnet test errors are given through the standard output stream.

## Environment
Powershell 7 on Windows 10

**Comment by nohwnd** [comment]
@RikkiGibson we had some requests for this, but can't figure out how to make the output colored while keeping it compatible with msbuild. Imho msbuild should be more of implementation detail than driver for the way we do output.

Or is this only about msbuild errors, and not all errors?

**Comment by RikkiGibson** [comment]
> Or is this only about msbuild errors, and not all errors?

I only observed the bad behavior with msbuild errors. For non-msbuild error messages it appears that the errors are correctly written out to the stderr stream.

Is there more that needs to be done for this issue aside from ensuring that msbuild error output is captured separately and forwarded to the appropriate stream?

**Comment by nohwnd** [comment]
This should be solved by #2702  and all the subsequent terminal logger work that is shipped in net9.
